File: 4st/Jisoo/programmer/올바른 괄호.java
class Solution {
    boolean solution(String s) {

        int count = 0;

        for (int i = 0; i < s.length(); i++) {

            if (s.charAt(i) == '(') {
                count++;
            } else {
                count--;
            }

            if (count < 0) {
                return false;
            }
        }

        return count == 0;
    }
}

File: 4st/Seohyun/programmer/올바른 괄호.js
function solution(s) {
  let stack = [];

  if (s.charAt(0) === ")") return false;

  for (const parens of s) {
    if (parens === "(") {
      stack.push(parens);
    } else {
      stack.pop();
    }
  }
  if (stack.length === 0) return true;

  return false;
}
